api/config.py

```diff
@@ -469,6 +479,53 @@ def configuration_contact_uri(cls, library):
             library, Configuration.CONFIGURATION_CONTACT_EMAIL
         )
 
+    @classmethod
+    def key_pair(cls, public_setting, private_setting):
+        """Look up a public-private key pair in two ConfigurationSettings.
+
+        If either setting is unset, a new key pair is created and
+        stored.
+
+        TODO: This could go into ConfigurationSetting or core Configuration.
+
+        :param public_setting: A ConfigurationSetting for the public key.
+        :param private_setting: A ConfigurationSetting for the private key.
+
+        :return: A 2-tuple (public key, private key)
+        """
+        # Prevent bad code where public and private key are mixed up.
+        if 'public' not in public_setting.key.lower():
```

Comment:
Why not check that it equals the constant?

Reply:
I thought this would be useful for storing other public/private key pairs, but we don't need that now.